modules/gdprEnforcement.js

```diff
@@ -191,11 +198,48 @@ export function userIdHook(fn, submodules, consentData) {
   }
 }
 
-const hasPurpose1 = (rule) => { return rule.purpose === purpose1 }
+/**
+ * Checks if the bidder is given consent. If yes, bid adapter is allowed to send ajax request to their endpoint, else, no request
+ * is sent. Enforces "purpose 2 (basic ads)" of TCF v2.0 spec
+ * @param {Function} fn - Function reference to the original function.
+ * @param {Array<adUnits>} adUnits
+ */
+export function makeBidRequestsHook(fn, adUnits, ...args) {
+  const consentData = gdprDataHandler.getConsentData();
+  if (consentData && consentData.gdprApplies) {
+    if (consentData.apiVersion === 2) {
+      const disabledBidders = [];
+      adUnits.forEach(adUnit => {
+        adUnit.bids = adUnit.bids.filter(bid => {
+          const currBidder = bid.bidder;
+          const gvlId = getGvlid(currBidder);
+          if (includes(disabledBidders, currBidder)) return false;
+          const isAllowed = validateRules(purpose2Rule, consentData, 2, currBidder, gvlId);
+          if (!isAllowed) {
+            utils.logWarn(`User blocked bidder: ${currBidder}. No bid request will be sent to their endpoint.`);
+            // Emit an event for the Analytics adapters to listen
```

Comment:
This comment is unnecessary, below line of code is self explainatory

Reply:
Removed comment.